src/decompress.rs

```diff
@@ -144,6 +144,65 @@ impl Decompressor {
             Err(unsafe { get_error(self.handle) })
         }
     }
+
+    /// Decompress a JPEG image in `jpeg_data` into `output` as YUV without changing color space.
+    ///
+    /// The decompressed image is stored in the pixel data of the given `output` image, which must
+    /// be fully initialized by the caller. Use [`read_header()`](Decompressor::read_header) to
+    /// determine the image size before calling this method.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// // read JPEG data from file
+    /// let jpeg_data = std::fs::read("examples/parrots.jpg")?;
+    ///
+    /// // initialize a decompressor
+    /// let mut decompressor = turbojpeg::Decompressor::new()?;
+    ///
+    /// // read the JPEG header
+    /// let header = decompressor.read_header(&jpeg_data)?;
+    /// // calculate yuv frame size
+    /// let yuv_frame_size = turbojpeg::yuv_frame_size(header.width, 4, header.height, header.subsamp);
+    ///
+    /// // initialize the image (YUVImage<Vec<u8>>)
+    /// let mut image = turbojpeg::YUVImage {
+    ///     pixels: vec![0; yuv_frame_size.unwrap()],
+    ///     width: header.width,
+    ///     pad: 4, // for video
+    ///     height: header.height,
+    /// };
+    ///
+    /// // decompress the JPEG into the image
+    /// // (we use as_deref_mut() to convert from &mut YUVImage<Vec<u8>> into YUVImage<&mut [u8]>)
+    /// decompressor.decompress_to_yuv(&jpeg_data, image.as_deref_mut())?;
+    /// assert_eq!(&image.pixels[0..4], &[116, 117, 118, 119]);
+    ///
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    #[doc(alias = "tjDecompressToYUV2")]
+    pub fn decompress_to_yuv(&mut self, jpeg_data: &[u8], output: YUVImage<&mut [u8]>) -> Result<()> {
+        let YUVImage { pixels, width, pad, height } = output;
+        let width = width.try_into().map_err(|_| Error::IntegerOverflow("width"))?;
+        let pad = pad.try_into().map_err(|_| Error::IntegerOverflow("pad"))?;
+        let height = height.try_into().map_err(|_| Error::IntegerOverflow("height"))?;
+        let jpeg_data_len = jpeg_data.len().try_into()
+            .map_err(|_| Error::IntegerOverflow("jpeg_data.len()"))?;
+        let res = unsafe {
+            raw::tjDecompressToYUV2(
+                self.handle,
+                jpeg_data.as_ptr(), jpeg_data_len,
+                pixels.as_mut_ptr(), width, pad, height,
+                0,
+            )
+        };
```

Comment:
Before calling `tjDecompressToYUV2()`, I think that we must verify that `pixels.len()` is at least as long as the value returned from `tjBufSizeYUV2()`, to make sure that the call is in fact safe.